hyperspy/_components/polynomial_deprecated.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2007-2016 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+import logging
+
+
+from hyperspy.component import Component
+from hyperspy._components.expression import Expression
+from hyperspy.misc.utils import ordinal
+from hyperspy.docstrings.parameters import FUNCTION_ND_DOCSTRING
+
+
+_logger = logging.getLogger(__name__)
+
+
+class Polynomial(Component):
+
+    """n-order polynomial component. (DEPRECATED)
+    Polynomial component defined by the coefficients parameters which is an
+    array of len the order of the polynomial.
+    For example, the [1,2,3] coefficients define the following 3rd order
+    polynomial: f(x) = 1x² + 2x + 3 `Polynomial` will be replaced by
+    `Polynomial2`
+
+    This API is deprecated and will be replaced by
+    :py:class:`hyperspy._components.polynomial.Polynomial` in HyperSpy v2.0.
+    To use the new API, set `legacy` to `False`.
+
+    Attributes
+    ----------
+    coeffcients : array
+    """
+
+    def __init__(self, order=2, legacy=True, module="numexpr", **kwargs):
+        """Polynomial component (DEPRECATED)
+
+        This API is deprecated and will be replaced by
+        :py:class:`hyperspy._components.polynomial.Polynomial` in HyperSpy v2.0.
+        To use the new API, set `legacy` to `False`.
+
+        Parameters
+        ----------
+        order: int
+            Order of the polynomial.
+        legacy: bool, default True
+            If `False`, use the new API.
+        module: str
+            See the docstring
+            of :py:class:`hyperspy._components.polynomial.Polynomial`
+            for details.
+        """
+        if legacy:
+            from hyperspy.misc.utils import deprecation_warning
+            msg = (
+                "The API of the `Polynomial` component will change in v2.0."
+                "To use the new API set `legacy=False`.")
+            deprecation_warning(msg)
+
+            Component.__init__(self, ['coefficients', ])
+            self._whitelist['order'] = ('init', order)
```

Comment:
I don't think this is necessary to add `legacy` to the whitelist, but I am mentioning it in case it has been overseen and I would have miss something.

Reply:
I agree.